tools/PI/DevHome.PI/Models/AppRuntimeInfo.cs

```diff
@@ -2,84 +2,157 @@
         FrameworkTypes.Add(new FrameworkType("PresentationFramework.dll", "WPF"));
     }
 
-    public void CheckFrameworkTypes(string moduleName)
+    public void CheckFrameworksAndCommandLine(Process process)
     {
-        foreach (var item in FrameworkTypes)
+        var identifiedFrameworks = new List<string>();
+        var modules = process.Modules;
+        foreach (ProcessModule module in modules)
         {
-            // Skip if already matched.
-            if (item.IsTypeSupported == true)
+            for (var i = 0; i < FrameworkTypes.Count; i++)
             {
-                continue;
-            }
+                var item = FrameworkTypes[i];
 
-            if (item.IsExactMatch)
-            {
-                if (moduleName.Equals(item.Determinator, StringComparison.OrdinalIgnoreCase))
+                if (identifiedFrameworks.Contains(item.Name))
                 {
-                    item.IsTypeSupported = true;
+                    continue;
                 }
-            }
-            else
-            {
-                if (moduleName.Contains(item.Determinator, StringComparison.OrdinalIgnoreCase))
+
+                if (item.IsExactMatch)
+                {
+                    if (module.ModuleName.Equals(item.Determinator, StringComparison.OrdinalIgnoreCase))
+                    {
+                        identifiedFrameworks.Add(item.Name);
+                    }
+                }
+                else
                 {
-                    item.IsTypeSupported = true;
+                    if (module.ModuleName.Contains(item.Determinator, StringComparison.OrdinalIgnoreCase))
+                    {
+                        identifiedFrameworks.Add(item.Name);
+                    }
                 }
             }
         }
+
+        // Both WinUI2 and WinUI3 use Microsoft.UI.Xaml.dll, but only WinUI3 uses Microsoft.UI.Xaml.Controls.dll.
+        if (identifiedFrameworks.Contains("WinUI 3"))
+        {
+            identifiedFrameworks.Remove("WinUI 2");
+        }
+
+        IdentifiedFrameWorkTypes = string.Join(", ", identifiedFrameworks);
+
+        // The only reliable check for UWP is if the command-line matches the known UWP pattern.
+        GetCommandLine(process);
+        if (UwpCommandLineRegex().IsMatch(ActivationArgs))
+        {
+            IdentifiedFrameWorkTypes += ", UWP";
+            ActivationArgs = _unknownText;
+        }
+    }
+
+    private void GetCommandLine(Process process)
+    {
+        try
+        {
+            using var searcher = new ManagementObjectSearcher(
+                $"SELECT CommandLine FROM Win32_Process WHERE ProcessId = {process.Id}");
+            if (searcher is null)
+            {
+                return;
+            }
+
+            using var objects = searcher.Get();
+            if (objects is null)
+            {
+                return;
+            }
+
+            var obj = objects.Cast<ManagementObject>().FirstOrDefault();
```

Comment:
Thinking out loud.  No need t6o respond to this.  Would reversing the operation produce faster code?  Get the first object, then cast it to a `ManagementObject`

If so, the improvement is negligible.